frame/balances/src/benchmarking.rs

```diff
@@ -173,7 +173,7 @@ mod benchmarks {
 	// but additionally introduces many new users into the storage, increasing the the merkle
 	// trie and PoV size.
 	#[benchmark(extra)]
-	fn transfer_increasing_users(u: Linear<0, 1_000>) {
+	fn transfer_increasing_users(u: linear![0..1_000]) {
```

Comment:
Why not keep `Linear<0..1_000>`?  
Since const generics cannot accept ranges or something?

Reply:
I think the advantage of having a type is that people can click on it in the language server and it explains what it does.

Reply:
the reason for the fake enclosing macro is rust isn't able to parse the enclosing module as an `ItemMod` if we do `Linear<0..10>` because the `0..10` can't appear in type position, so we would either have to do some kind of pre-pass where we remove these before parsing as an `ItemMod` (hard) or enclose the range in something that works in any position like a macro call

Reply:
So based on those trade-offs, I like this

Reply:
The whole root of this issue is if we are supporting things that are `Into<u32>` instead of just `u32` literals, this is no longer really a type, it's an expression

Reply:
maybe something with a tuple though......?

Reply:
Well, we can use `Into<u32>` but only with the `Linear<start, end>` syntax and not `…`.
I dont see anything wrong with `Linear<start, end>` besides the fact that it is "less" rust.  
On the other hand the `start…end` is also not quite right in the rust semantic, since we just skip specific values depending on the `step` size. So it is not actually a range in the rust sense.

Reply:
But it is still a valid rust type; just needs brackets: `Linear<{ something.something() }, 33>`. Everything we do here must be compile time anyway, but that also applies to the V1 syntax.

This works just fine:
https://github.com/paritytech/substrate/blob/oty-demo-bench-syntax/frame/examples/basic/src/benchmarking.rs#L105

Reply:
Interesting! Thanks @ggwpez. Let me take this up.